ingest/src/main/scala/hydra.ingest/http/TopicDeletionEndpoint.scala

```diff
@@ -31,28 +31,45 @@ object TopicDeletionEndpoint extends
 
   final case class DeletionEndpointResponse(success:List[SuccessfulResponse], clientError: List[ErroredResponse], serverError: List[ErroredResponse])
 
-  final case class DeletionRequest(topics: List[String], ignoreConsumerGroups: List[String], ignorePublishTime: Boolean)
+  final case class DeletionRequest(topics: List[String], ignoreConsumerGroups: List[String], ignorePublishTime: Boolean, ignoreAllConsumerGroups: Boolean)
 
   implicit object DeletionRequest extends RootJsonFormat[DeletionRequest] {
     override def read(json: JsValue): DeletionRequest = {
-      json.asJsObject.getFields("topics", "ignoreConsumerGroups", "ignorePublishTime") match {
-        case Seq(topics, ignoreConsumerGroups, ignorePublishTime) =>
-          DeletionRequest(topics.convertTo[List[String]], ignoreConsumerGroups.convertTo[List[String]], ignorePublishTime.convertTo[Boolean])
-        case Seq(topics, ignoreSomething) =>
-          try {
-            DeletionRequest(topics.convertTo[List[String]], ignoreSomething.convertTo[List[String]], false)
-          } catch {
-            case e: Exception => DeletionRequest(topics.convertTo[List[String]], List.empty, ignoreSomething.convertTo[Boolean])
-          }
-        case Seq(topics) =>
-          DeletionRequest(topics.convertTo[List[String]], List.empty, false)
-        case _ =>
-          spray.json.deserializationError("Must provide a List of topics to delete")
+      def convertFromArrayToList(valueOpt: Option[JsValue]): List[String] =
+        valueOpt match {
+          case Some(JsArray(value)) => value.map(_.convertTo[String]).toList
+          case _ => List.empty
+        }
+
+      def parseBooleanOpt(valueOpt: Option[JsValue]): Boolean =
+        valueOpt match {
+          case Some(JsBoolean(value)) => value
+          case _ => false
+        }
+
+      val allFields = json.asJsObject.fields
+
+      val topics = convertFromArrayToList(allFields.get("topics"))
+      val ignoreConsumerGroups = convertFromArrayToList(allFields.get("ignoreConsumerGroups"))
+      val ignorePublishTime = parseBooleanOpt(allFields.get("ignorePublishTime"))
+      val ignoreAllConsumerGroups = parseBooleanOpt(allFields.get("ignoreAllConsumerGroups"))
+
+      if (topics.isEmpty) {
+        spray.json.deserializationError("Must provide a List of topics to delete")
+      } else {
+        DeletionRequest(topics, ignoreConsumerGroups, ignorePublishTime, ignoreAllConsumerGroups)
```

Comment:
Why not use reduced JsonFormats for case clasess here? (https://github.com/spray/spray-json#providing-jsonformats-for-case-classes)